src/commands/notify.yml

```diff
@@ -134,6 +135,10 @@ steps:
           equal: [ "", <<parameters.thread_id>> ]
       steps:
         - save_cache:
-            key: cache-<< parameters.thread_id >>-{{ .Environment.CIRCLE_WORKFLOW_ID }}
+            key: cache-{{ .Revision }}-{{ .Environment.CIRCLE_WORKFLOW_ID }}
+            paths:
+              - /tmp/SLACK_THREAD_INFO
+        - save_cache:
+            key: cache-{{ .Revision }}
             paths:
               - /tmp/SLACK_THREAD_INFO
```

Comment:
I think just the first `save_cache` is required

Reply:
I see your point, but the goal here is to search for an explicit fallback cache. If I remove the second `save_cache` and change the second key in `restore_cache` by adding a `-` at the end, the rerun pipeline won't get to the fallback cache key because there won't be such, because `{{ .Environment.CIRCLE_WORKFLOW_ID }}` is never going to be empty, and `cache-{{ .Revision }}-` will never be initiated.

Reply:
@atanass our docs do not seem to be as clear as they really should be on the matter, so I'll address that separately, but keys are used in a sort of prefix search.

So by creating the key `cache-{{ .Revision }}-{{ .Environment.CIRCLE_WORKFLOW_ID }}` we are doing enough already, since a prefix search using `cache-{{ .Revision }}` will find the _most recent_ matching key with that prefix.

Does that make sense?

If so, I'll raise a separate issue over on the docs to update https://circleci.com/docs/caching/#restoring-cache

Reply:
Actually maybe my eyesight is the problem rather than the docs, as that page does mention it albeit it's subtle `Each cache key is namespaced to the project and retrieval is prefix-matched`

Reply:
I see your point. I think you're right. Let me test that and will update accordingly.
Thank you.

Reply:
We also _just_ made an update that could make this even simpler, as we now have the environment variable `CIRCLE_PIPELINE_ID` available. This ID is the same between re-runs, and could replace the revision/workflow ID combo.

Reply:
Note I think this was literally just added today, so technically would not work for any pipelines created before today.

Reply:
Brilliant. That's exactly what I needed! Pushing an updated version.